tools/osx/macosx_bundle.sh

```diff
@@ -110,7 +110,9 @@ EXPATLIB="$(cmake .. -LA -N | grep "pkgcfg_lib_EXPAT_expat" | cut -d "=" -f2)"
 CODESIGNID="$(cmake .. -LA -N | grep "CODESIGNID" | cut -d "=" -f2)"
```

Comment:
Would be good if you added a comment showing the output of the first command (e.g. the output of `cmake .. -LA -N`) and then the expected final result (`$CODESIGNID`).

Reply:
Actually the reason cmake is called is to show all the cached variables. Could there be a more proper way of getting cmake-cached variables.

Reply:
@Benitoite looks like you could use `configure_file`:
https://cmake.org/cmake/help/v3.8/command/configure_file.html

So you could have a section in the Bash script with a list of `@VARIABLES@` as such:
```bash
codesign_id=@CODESIGNID@
```
and then in CMake run:
```cmake
configure_file(macosx_bundle.sh.in macosx_bundle.sh @ONLY)
```

Otherwise, if you prefer to `grep` then you could do it as you do now, with one small change. Since you're already using Bash then I would stick to Bash, so instead of calling `cut` (which is not part of Bash) I would use Bash's substring removal (here I will use `CMAKE_BUILD_TYPE` as an example):
```bash
# In: CMAKE_BUILD_TYPE:STRING=release
# Out: release
foo="$(cmake .. -L -N | grep CMAKE_BUILD_TYPE)"; foo="${foo#*=}"
```

Reply:
Thanks, substring removal is the best way with grep, which only needs to keep credentials in the cache.